pkg/log/config.go

```diff
@@ -50,3 +51,108 @@ debug logs.
 Such as you can enable 'gossip' logs with '--log.subsystems gossip'.`,
 	)
 }
+
+type AccessLogHeaderConfig struct {
+	// Prevent these headers from being logged.
+	// You can only define one of Allowlist or Blocklist.
+	Blocklist []string `json:"blocklist" yaml:"blocklist"`
+
+	// Log only these headers.
+	// You can only define one of Allowlist or Blocklist.
+	Allowlist []string `json:"allowlist" yaml:"allowlist"`
+
+	// In-memory map that is composed right after arguments have
+	// been validated. This is used to speed up runtime lookups
+	// for headers.
+	allowList map[string]string
+}
+
+func (c *AccessLogHeaderConfig) Validate() error {
+	if len(c.Allowlist) > 0 && len(c.Blocklist) > 0 {
+		return fmt.Errorf("cannot define both allowlist and blocklist")
+	}
+
+	// Create the allow-list header map used to filter headers at runtime.
+	if len(c.Allowlist) > 0 {
+		c.allowList = make(map[string]string)
+		for _, el := range c.Allowlist {
+			c.allowList[el] = el
+		}
+	}
+	return nil
+}
+
+func (c *AccessLogHeaderConfig) Filter(h http.Header) http.Header {
```

Comment:
keen to avoid "logic" in the config

what about moving this (along with the `allowList` cache) into `middleware.NewLogger`. such as add `middleware.LogFilter`, which can also then be unit tested properly

Reply:
Sounds good, will do!

Reply:
I haven't exactly made the filtering its own middleware, but I have co-located it with `middleware.NewLogger(..)`: https://github.com/andydunstall/piko/pull/230/commits/2e074aaf3d38411ed2ed48f82c407f51af0941b8

Please let me know if this change makes sense to you! I wasn't quite sure how to structure the filter into its own middleware that would then be used everywhere we called `middleware.NewLogger(..)`, hence this approach, as the filtering needs to happen at the time of the request itself being logged